app/serializers/application_serializer.rb

```diff
@@ -16,4 +17,8 @@ def string_formatter
   def number_formatter
     @number_formatter ||= NumberFormatter.new
   end
+
+  def date_formatter
+    @date_formatter ||= DateFormatter.new
+  end
```

Comment:
This is a common pattern in Ruby typically called "memoization" and it sets `@date_formatter` only if its "falsy" (`nil` or `false`).